### PR TITLE
koord-scheduler: implement PreBindExtensions to support one-time apply patches

### PR DESCRIPTION
**Pull request by eahydra**
### Ⅰ. Describe what this PR does

Optimize the Patch behavior of the koord-scheduler custom plugins:
1. Added the `DefaultPrebind` plugin to initiate a Patch request. The plugin implements the new interface `PreBindExtensions`. 
2. `DefaultPreBind` will be added to `KubeSchedulerConfig` by default. The plugin can be disabled by setting the `Disabled` parameter, but after disabling, a new plugin implementation of `PreBindExtensions` needs to be provided.
3. Custom plugins such as NodeNUMAResources and DeviceShare only modify Pods during PreBind and do not initiate Patch requests.
4. In this way, multiple Patch requests can be converted into one. And it can support custom implementation of different Patch strategies.

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Review by saintube — COMMENTED** [review]
/lgtm

File: pkg/scheduler/frameworkext/framework_extender.go [hunk 1]
```diff
@@ -270,6 +283,8 @@ func (ext *frameworkExtenderImpl) RunPreBindPlugins(ctx context.Context, state *
 	reservation = reservation.DeepCopy()
 	reservation.Status.NodeName = nodeName
 
+	original := reservation
+	reservation = reservation.DeepCopy()
```
Comment (FillZpp) [inline pkg/scheduler/frameworkext/framework_extender.go:287]: Is there any way to avoid deepcopy the reservation twice?

Comment (eahydra) [reply pkg/scheduler/frameworkext/framework_extender.go:284]: updated

**Review by FillZpp — APPROVED** [review]
/lgtm

**Review by buptcozy — APPROVED** [review]
/arrove

**Pull request merged**